src/main/python/systemds/matrix/matrix.py

```diff
@@ -22,22 +22,29 @@
 __all__ = ['Matrix', 'federated', 'full', 'seq']
 
 import os
-from typing import Union, Optional, Iterable, Dict, Tuple, Sequence
+from typing import Union, Optional, Iterable, Dict, Tuple, Sequence, TYPE_CHECKING
 
 import numpy as np
 from py4j.java_gateway import JVMView, JavaObject
 
 from systemds.utils.converters import numpy_to_matrix_block
-from systemds.script_building.dag import VALID_INPUT_TYPES
 from systemds.matrix.operation_node import OperationNode
 
+from systemds.utils.consts import VALID_INPUT_TYPES
+
+if TYPE_CHECKING:
+    # so we don't get cyclic dependencies during runtime
```

Comment:
to avoid cyclic dependencies during runtime from SystemDSContext

Reply:
I don't know why I used "we" so much, maybe I got a bit lonely during the implementation :smile: